ledger-core/v2/conveyor/conveyor.go

```diff
@@ -356,9 +356,9 @@ func (p *PulseConveyor) CommitPulseChange(pr pulse.Range) error {
 			case p.getPulseSlotMachine(prevPresentPN) != p.presentMachine:
 				panic("illegal state")
 			case pr.LeftBoundNumber() != prevFuturePN:
-				panic("illegal state")
+				panic(fmt.Sprintf("illegal state: leftbound %v, expected %v", prevFuturePN, pr.LeftBoundNumber()))
```

Comment:
for the future, please write something more informative in panics. It's not clear what actually went wrong from the code.

Reply:
it depends. 
And also - we have to have housekeeping in the plans - to use structs for panics, not fmt 
@ita-sammann

Reply:
@sexyhamster pls add a jira+todo here, we have to do it systematically, not at spot

Reply:
not sure about whrere to place TODO. but I've created PLAT-93.